onnxruntime/python/tools/transformers/onnx_model_bert.py

```diff
@@ -387,7 +388,7 @@ def optimize(self, options: Optional[FusionOptions] = None, add_dynamic_axes: bo
 
         if options is not None:
             self.attention_mask.set_mask_format(options.attention_mask_format)
-            if options.use_multi_head_attention:
+            if options.use_multi_head_attention and not isinstance(self.attention_fusion, FusionBartAttention):
```

Comment:
This logic shall handle in onnx_model_bart.py. Why shall base class know about child class detail?

Reply:
The current `--use_multi_head_attention` flag logic replaces all attention nodes with multi-head attention nodes. However, both types of nodes exist in Whisper. This logic was added as a temporary workaround to maintain the existing flag logic. I think the flag should instead be used to enable multi-head attention fusion.